versioned_hdf5/tests/test_api.py

```diff
@@ -636,7 +636,7 @@ def test_timestamp_manual(vfile):
 def test_timestamp_pytz(vfile):
     # pytz is not a dependency of versioned-hdf5, but it is supported if it is
     # used.
-    import pytz
+    pytz = importorskip("pytz")
```

Comment:
I think we can probably drop support for `pytz`. We're using Python 3.10/3.11 and `zoneinfo`.

Reply:
I'm happy to do that. What about h5py? The versions being tested on CI are quite old.

Reply:
Dropping support for `h5py==2.10` after a discussion with @ArvidJB.